File: frontend/lib/features/accounting/domain/entities/accounting_voucher.dart
import 'accounting_voucher_line.dart';

class AccountingVoucher {
  const AccountingVoucher({
    required this.id,
    required this.workspaceId,
    required this.voucherNo,
    required this.voucherDate,
    required this.period,
    required this.status,
    required this.description,
    required this.totalDebit,
    required this.totalCredit,
    required this.createdByUserId,
    required this.createdAt,
    required this.updatedAt,
    required this.lines,
    this.postedAt,
  });

  final String id;
  final String workspaceId;
  final String voucherNo;
  final String voucherDate;
  final String period;
  final String status;
  final String description;
  final double totalDebit;
  final double totalCredit;
  final String createdByUserId;
  final String createdAt;
  final String updatedAt;
  final String? postedAt;
  final List<AccountingVoucherLine> lines;

  bool get isDraft => status == 'DRAFT';
}

File: backend/src/main/java/com/microflow/accounting/domain/model/AccountingVoucher.java
package com.microflow.accounting.domain.model;

import java.math.BigDecimal;
import java.util.List;

public record AccountingVoucher(
        String id,
        String workspaceId,
        String voucherNo,
        String voucherDate,
        String period,
        String status,
        String description,
        BigDecimal totalDebit,
        BigDecimal totalCredit,
        String createdByUserId,
        String createdAt,
        String updatedAt,
        String postedAt,
        List<AccountingVoucherLine> lines
) {
}
